linkis-engineconn-plugins/repl/src/main/java/org/apache/linkis/engineplugin/repl/executor/ReplEngineConnExecutor.java

```diff
@@ -0,0 +1,240 @@
+
+  private Map<String, String> configMap = new HashMap<>();
+
+  public ReplEngineConnExecutor(int outputPrintLimit, int id) {
+    super(outputPrintLimit);
+    this.id = id;
+  }
+
+  @Override
+  public void init() {
+    super.init();
+  }
+
+  @Override
+  public ExecuteResponse execute(EngineConnTask engineConnTask) {
+    Optional<Label<?>> userCreatorLabelOp =
+        Arrays.stream(engineConnTask.getLables())
+            .filter(label -> label instanceof UserCreatorLabel)
+            .findFirst();
+    Optional<Label<?>> engineTypeLabelOp =
+        Arrays.stream(engineConnTask.getLables())
+            .filter(label -> label instanceof EngineTypeLabel)
+            .findFirst();
+
+    if (userCreatorLabelOp.isPresent() && engineTypeLabelOp.isPresent()) {
+      UserCreatorLabel userCreatorLabel = (UserCreatorLabel) userCreatorLabelOp.get();
+      EngineTypeLabel engineTypeLabel = (EngineTypeLabel) engineTypeLabelOp.get();
+
+      Map<String, String> cacheMap =
+          new ReplEngineConf().getCacheMap(new Tuple2<>(userCreatorLabel, engineTypeLabel));
+      if (MapUtils.isNotEmpty(cacheMap)) {
+        configMap.putAll(cacheMap);
+      }
+    }
+
+    Map<String, Object> taskParams = engineConnTask.getProperties();
+
+    if (MapUtils.isNotEmpty(taskParams)) {
+      taskParams.entrySet().stream()
+          .filter(entry -> entry.getValue() != null)
+          .forEach(entry -> configMap.put(entry.getKey(), String.valueOf(entry.getValue())));
+    }
+
+    String replType = ReplConfiguration.REPL_TYPE.getValue(configMap);
+
+    replAdapter = ReplAdapterFactory.create(replType);
+    return super.execute(engineConnTask);
+  }
+
+  @Override
+  public ExecuteResponse executeLine(EngineExecutionContext engineExecutorContext, String code) {
+    String realCode;
+    if (StringUtils.isBlank(code)) {
+      throw new ReplException(
+          ReplErrorCodeSummary.REPL_CODE_IS_NOT_BLANK.getErrorCode(),
+          ReplErrorCodeSummary.REPL_CODE_IS_NOT_BLANK.getErrorDesc());
+    } else {
+      realCode = code.trim();
+    }
+    logger.info("Repl engine begins to run code:\n {}", realCode);
+
+    String taskId = engineExecutorContext.getJobId().get();
+
+    initialStatusUpdates(taskId, engineExecutorContext);
+
+    String classpathDir = ReplConfiguration.CLASSPATH_DIR.getValue(configMap);
+
+    String methodName = ReplConfiguration.METHOD_NAME.getValue(configMap);
+
+    Thread thread =
+        new Thread(
+            () -> {
+              try {
+                replAdapter.executorCode(realCode, classpathDir, methodName);
+              } catch (Exception e) {
+                String errorMessage = ExceptionUtils.getStackTrace(e);
+                logger.error("Repl engine execute failed : {}", errorMessage);
+                engineExecutorContext.appendStdout(LogUtils.generateERROR(errorMessage));
+              }
+            });
+
+    thread.start();
+    threadCache.put(taskId, thread);
```

Comment:
Is there no need to be asynchronous here? Just use the current thread and interrupt the thread when killing.